File: 1-Two-Sum/1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_map = {}
        for i,n in enumerate(nums):
            diff = target - n
            if diff in num_map:
                return [num_map[diff], i]
            num_map[n] = i

File: 1-Two-Sum/1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let numMap = new Map();
    for(let i = 0; i < nums.length; i++){
        let diff = target - nums[i];
        if(numMap.has(diff)){
            return [i, numMap.get(diff)];
        }
        numMap.set(nums[i], i);
    }
};